packages/expo-modules-core/android/src/test/java/expo/modules/kotlin/methods/SuspendMethodTest.kt

```diff
@@ -0,0 +1,156 @@
+package expo.modules.kotlin.methods
+
+import com.facebook.react.bridge.JavaOnlyArray
+import com.google.common.truth.Truth
+import expo.modules.PromiseMock
+import expo.modules.PromiseState
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.async
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class SuspendMethodTest {
+  @Test
+  fun `suspend block should resolve promise on finish`() = runTest {
+    val suspendMethod = SuspendMethod("test", emptyArray(), lazy { this }) {
+      delay(2000)
+    }
+    val promiseMock = PromiseMock()
+
+    suspendMethod.call(JavaOnlyArray(), promiseMock)
+    testScheduler.advanceTimeBy(3000)
+
+    Truth.assertThat(promiseMock.state).isEqualTo(PromiseState.RESOLVED)
+  }
+
+  @Test
+  fun `suspend block should reject promise when throws`() = runTest {
+    val suspendMethod = SuspendMethod("test", emptyArray(), lazy { this }) {
+      delay(2000)
+      throw IllegalStateException()
+    }
+    val promiseMock = PromiseMock()
+
+    suspendMethod.call(JavaOnlyArray(), promiseMock)
+    testScheduler.advanceTimeBy(3000)
+
+    Truth.assertThat(promiseMock.state).isEqualTo(PromiseState.REJECTED)
+  }
```

Comment:
Could you check if this test passes? 🤔  See my other comment above
```suggestion
  }
  
  @Test
  fun `suspend block should reject when nested coroutine throws`() = runTest {
    val suspendMethod = SuspendMethod("test", emptyArray(), lazy { this }) {
      launch {
        delay(2000)
        throw IllegalStateException()
      }
    }
    val promiseMock = PromiseMock()

    suspendMethod.call(JavaOnlyArray(), promiseMock)
    testScheduler.advanceTimeBy(3000)

    Truth.assertThat(promiseMock.state).isEqualTo(PromiseState.REJECTED)
  }
```